src/game/counting.rs

```diff
@@ -0,0 +1,276 @@
+use crate::game::cards::Card;
+use crate::game::scoring::{score_run, Combination, CombinationKind, Score};
+use std::error::Error;
+
+/// Calculates the score during the counting phase where played_cards have already been played and card is now played
+///
+///
+pub fn score_counting_cards_played(
+    played_cards: &[Card],
+    card: Card,
+) -> Result<Score, Box<dyn Error>> {
```

Comment:
if it isn't boxed (-> Result<Score, dyn Error>), I get this error

> the size for values of type `(dyn std::error::Error + 'static)` cannot be known at compilation time
> the trait `std::marker::Sized` is not implemented for `(dyn std::error::Error + 'static)`rustcE0277
> lib.rs(1, 1): required by a bound in this
> result.rs(241, 20): required by this bound in `std::result::Result`

I suspect there are two ways to fix this -- Box it so that the compiler ignores the error (my hope is that the Box gives it a specific size, so it all works as intended) or to build a custom Error (CribbageError?) that the compiler can tell its size.

Reply:
If you plan to use the error for flow control/UX at higher levels, it's worth making a specific error type.